gfx/common/x11_common.c

```diff
@@ -306,9 +306,17 @@ static void x11_suspend_screensaver_xdg_screensaver(Window wnd, bool enable)
 
 void x11_suspend_screensaver(Window wnd, bool enable)
 {
-    x11_suspend_screensaver_xdg_screensaver(wnd, enable);
 #ifdef HAVE_DBUS
     x11_suspend_screensaver_dbus(enable);
+    if (dbus_screensaver_cookie == 0)
+    {
+        // Only attempt xdg-screensaver method if dbus didn't succeed
+#endif
+
+        x11_suspend_screensaver_xdg_screensaver(wnd, enable);
+
+#ifdef HAVE_DBUS
+    }
```

Comment:
> dbus uninhibiting is unimplemented

Ah, but I think it is: https://github.com/libretro/RetroArch/blob/master/gfx/common/x11_common.c#L720 That was part of my previous PR to implement DBus support.

Perhaps the confusion comes from the fact that `x11_suspend_screensaver()` is actually only ever [called once on video init](https://github.com/libretro/RetroArch/blob/master/gfx/video_driver.c#L751) I believe. For example toggling the "suspend screensaver" setting on the menu doesn't call this function AFAIK, so there's no need for additional smarts in `x11_suspend_screensaver()`.

> But this is not a DBus method. It's the generic 'suspend X11 screensaver by any means necessary' function

True, but that's inside an `#ifdef HAVE_DBUS` though. Other functions have been updated with similar blocks to call DBus-specific code (see link above in `x11_window_destroy` for instance). 

To me it seems referencing a dbus function inside such blocks (e.g. `if (dbus_inhibited())`) vs. referencing a variable (e.g. `if (dbus_cookie == 0)` is the same, but I guess I don't mind wrapping it in a function if that's your preference.